File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.js
/**
 * @param {number[]} nums
 * @param {number} maximumBit
 * @return {number[]}
 */
var getMaximumXor = function(nums, maximumBit) {
    let n = nums.length
    let b = (1 << maximumBit) - 1
    const arr = Array(n).fill(0)
    let xor = 0

    for(let num of nums) 
        xor ^= num
    
    for (let i = 0; i < n; ++i) {
        arr[i] = xor ^ b
        xor ^= nums[n - i - 1]
    }
    return arr
};

File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.cs
public class Solution {
    public int[] GetMaximumXor(int[] nums, int maximumBit) {
        int n = nums.Length;
        int b = (1 << maximumBit) - 1;
        int[] res = new int[n];
        int xor = 0;

        foreach(int num in nums)
            xor ^= num;
        
        for (int i = 0; i < n; ++i) {
            res[i] = xor ^ b;
            xor ^= nums[n - i - 1];
        }
        return res;
    }
}